library/x509_crt.c

```diff
@@ -1605,21 +1605,23 @@ int mbedtls_x509_crt_check_extended_key_usage( const mbedtls_x509_crt *crt,
  */
 int mbedtls_x509_crt_is_revoked( const mbedtls_x509_crt *crt, const mbedtls_x509_crl *crl )
 {
-    const mbedtls_x509_crl_entry *cur = &crl->entry;
+    mbedtls_x509_crl_entry **entry, *key;
+    int ret = 0;
 
-    while( cur != NULL && cur->serial.len != 0 )
-    {
-        if( crt->serial.len == cur->serial.len &&
-            memcmp( crt->serial.p, cur->serial.p, crt->serial.len ) == 0 )
-        {
-            if( mbedtls_x509_time_is_past( &cur->revocation_date ) )
-                return( 1 );
-        }
+    key = mbedtls_calloc(1, sizeof( *key ) );
```

Comment:
or I can simply use an auxiliary `mbedtls_x509_crl_entry *key_ptr` variable to store the address of the `key` variable allocated on the stack. Then I can pass `&key_ptr` to bsearch().
Not extremely pretty, but this way we can avoid the even uglier calloc/free invocation.